Configuration warning on stable versions when running test suite

I noticed the following issue on Travis for v4.0.x - the **Python 3.6 with oldest supported version of all dependencies** build fails when pushing the commit that contains a stable (non dev) version:

https://travis-ci.org/github/astropy/astropy/jobs/734636060

The warning, made into an error, is:

```
ImportError while loading conftest '/home/travis/build/astropy/astropy/conftest.py'.
conftest.py:18: in <module>
    import astropy
astropy/__init__.py:371: in <module>
    _initialize_astropy()
astropy/__init__.py:298: in _initialize_astropy
    config.configuration.update_default_config(__package__, config_dir)
astropy/config/configuration.py:738: in update_default_config
    ConfigurationChangedWarning)
E   astropy.config.configuration.ConfigurationChangedWarning: The configuration options in astropy 4.0.2 may have changed, your configuration file was not updated in order to preserve local changes.  A new configuration template has been saved to '/home/travis/.astropy/config/astropy.4.0.2.cfg'.
```

I never noticed this because often that commit is pushed once the release branch CI is otherwise passing. I tried clearing the Travis cache to no avail.

--- a/.travis.yml
+++ b/.travis.yml
@@ -113,6 +113,7 @@ jobs:
           env: PYTHON_VERSION=3.6 NUMPY_VERSION=1.16
                CONDA_DEPENDENCIES=$CONDA_ALL_DEPENDENCIES
                SCIPY_VERSION=0.18 MATPLOTLIB_VERSION=2.1
+               PYTEST_VERSION=4
                INSTALL_CMD='python setup.py build_ext --inplace'
                PIP_DEPENDENCIES='pytest-astropy pytest-openfiles==0.4.0'
                TEST_CMD='pytest --open-files --doctest-rst'

